src/app.types.tsx

```diff
@@ -124,6 +124,7 @@ export interface ShotnumRange {
 export interface SearchParams {
   dateRange: DateRange;
   shotnumRange: ShotnumRange;
+  maxShots: number | 'Unlimited';
```

Comment:
I wonder if some of the logic here would be simplified if "Unlimited" `maxShots` is represented by `Infinity`. It would mean no special handling for the table - as `totalDataCount < maxShots ? totalDataCount : maxShots` works if `maxShots` is `Infinity`. Also means we don't have to deal with both numbers and strings in parsing the values in `maxShots.component.tsx` - the only change is instead of using `parseInt(value)` we'd have to use `Number(value)`. Thoughts? Feel free to ignore if you try it and the logic is more complex

Reply:
Use of `Infinity` had not crossed my mind but I like it and it does simplify things. I'll make the change